src/dexpo.cpp

```diff
@@ -61,39 +61,107 @@ main ()
   window w (dexpo_x, dexpo_y, conf_width, conf_height);
   w.pixmaps = v;
   // Mapping pixmap onto window
-  while (1)
+  int highlighted = 0;
+  xcb_generic_event_t *event;
+  while ((event = xcb_wait_for_event (window::c_)))
     {
-      usleep (1000);
-
-      if (dexpo_width == 0)
+      switch (event->response_type & ~0x80)
         {
-          // Drawing screenshots starting from the left top corner
-          auto act_width = dexpo_padding;
-          for (const auto &dexpo_pixmap : v)
+        case XCB_EXPOSE:
+          {
+            if (dexpo_width == 0)
+              {
+                // Drawing screenshots starting from the left top corner
+                auto act_width = dexpo_padding;
+                for (const auto &dexpo_pixmap : v)
+                  {
+                    xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
+                                   desktop_pixmap::gc_, 0, 0, act_width,
+                                   dexpo_padding, dexpo_pixmap.width,
+                                   dexpo_pixmap.height);
+                    act_width += dexpo_pixmap.width;
+                    act_width += dexpo_padding;
+                  };
+              }
+            else if (dexpo_height == 0)
+              {
+                auto act_height = dexpo_padding;
+                for (const auto &dexpo_pixmap : v)
+                  {
+                    xcb_copy_area (window::c_, dexpo_pixmap.id, w.id,
+                                   desktop_pixmap::gc_, 0, 0, dexpo_padding,
+                                   act_height, dexpo_pixmap.width,
+                                   dexpo_pixmap.height);
+                    act_height += dexpo_pixmap.height;
+                    act_height += dexpo_padding;
+                  };
```

Comment:
Do you think this should be reformatted and moved somewhere?
Looks like spaghetti

Reply:
Shouldn't vector refresh in cycle, so screenshots'll be more or less actual?